### PR TITLE
Sprint7 task1 photo to reward

### PR DESCRIPTION
**Pull request by iAcay**
Hey! The code which is below was implemented in order to meet the acceptance criterias of the first task in sprint no. 7 :)

AC in this task:
-- An admin can attach a .jpg or .png file to a reward
-- There is a file format validation that allows upload only .jpg or .png
-- A user can see the uploaded reward photo
-- Photos are being persisted after consequent deploys to Heroku

I use AWS S3 as 3rd party service for storage uploaded photos :) 

https://user-images.githubusercontent.com/96024046/182833631-6caf6420-c900-4032-aa9a-aee7d8ebde47.mov

### CODE REVIEW

File: app/controllers/AdminUsers/rewards_controller.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 module AdminUsers
   class RewardsController < BaseController
     def index
-      render :index, locals: { rewards: Reward.order(title: :asc).includes([:category]) }
+      render :index, locals: { rewards: Reward.order(title: :asc).includes([:category, [photo_attachment: :blob]]) }
```
Comment (piowit) [inline app/controllers/AdminUsers/rewards_controller.rb]:
```suggestion
      rewards: Reward.order(title: :asc).with_attached_photos.includes([:category]) }
```
There is method to help fix n+1 while attaching photos that looks better than this include :)
https://api.rubyonrails.org/classes/ActiveStorage/Attached/Model.html#method-i-has_many_attached

**Review by piowit — COMMENTED** [review]
🔥 Looks good, left one idea for change below

File: app/controllers/AdminUsers/rewards_controller.rb (hunk 1 above)
Comment (iAcay) [reply app/controllers/AdminUsers/rewards_controller.rb]: Thanks a lot! 👍  I changed this :)

**Review by maikhel — APPROVED** [review]
Looks great 👍 

If you want, you can try to add system test for file upload

**Comment by iAcay** [comment]
> Looks great 👍
> 
> If you want, you can try to add system test for file upload

Thanks a lot for CR! :) On your advice I have added system spec for photo uploading by admin 👍

**Pull request merged**